plugin/src/operations.ts

```diff
@@ -126,39 +126,61 @@ export function createOperations(
    */
   async function completedOperation(
     operationId: string,
+    nodeStatsChangedCallback: (node: BulkOperationNode) => void,
     interval = 1000
   ): Promise<{ node: BulkOperationNode }> {
-    const operation = await client.request<{
-      node: BulkOperationNode;
-    }>(OPERATION_BY_ID, {
-      id: operationId,
-    });
+    let operation = await client.request<{
+        node: BulkOperationNode;
+      }>(OPERATION_BY_ID, {
+        id: operationId,
+      });
 
-    if (options.verboseLogging) {
-      reporter.verbose(`
-        Waiting for operation to complete
+    nodeStatsChangedCallback(operation.node);
+    
+    while (true) {
+      if (options.verboseLogging) {
+        reporter.verbose(`
+          Waiting for operation to complete
 
-        ${operationId}
+          ${operationId}
 
-        Status: ${operation.node.status}
+          Status: ${operation.node.status}
 
-        Object count: ${operation.node.objectCount}
+          Object count: ${operation.node.objectCount}
 
-        Url: ${operation.node.url}
-      `);
-    }
+          Url: ${operation.node.url}
+        `);
+      }
 
-    if (failedStatuses.includes(operation.node.status)) {
-      throw new OperationError(operation.node);
-    }
+      if (failedStatuses.includes(operation.node.status)) {
+        throw new OperationError(operation.node);
+      }
 
-    if (operation.node.status === "COMPLETED") {
-      return operation;
-    }
+      if (operation.node.status === "COMPLETED") {
+        return operation;
+      }
+
+      await new Promise((resolve) => setTimeout(resolve, interval));
+
+      const nextOperation = await client.request<{
+        node: BulkOperationNode;
+      }>(OPERATION_BY_ID, {
+        id: operationId,
+      });
 
-    await new Promise((resolve) => setTimeout(resolve, interval));
+      const operationNodeFields = Object.keys(operation.node) as (keyof BulkOperationNode)[];
+      let updated: boolean = false;
```

Comment:
So I think I just realized something: I don't think we should care if it's been updated or not. It seems like there's absolutely no harm in just unconditionally calling `setStatus`, because it doesn't create any additional noise in the output.

Reply:
I had thought of that before, but I think this route is ~safer~ in the long run just incase down the road the callback that sets the status has some other functionality as well that may cause additional noise or unnecessary computation. This way we don't have to worry about how the callback changes, we know it's only getting called when there is a change. I don't have strong opinions about it though, happy to go about it the simpler route as well.

Reply:
Yeah I hear you. Another thing I was thinking about is, if we really want to check for updates, it might be easier to do this:

```ts
const updated = JSON.stringifiy(operation.node) !== JSON.stringify(nextOperation.node)
```

That's safe as long as the object's keys will always be in the same order, which in this case they surely would be since the API would always be serializing its data in the same way.

Reply:
I like it, the only thing I'm weary about that is if Shopify changes its API, let's say they add a timestamp or something that will be different for every API call. Or they adds tons of different fields in it and we need to process a bunch of fields that we aren't actually using, that may add extra overhead.

If we keep it to the original implementation we control the fields that we want to be processed (it will just be whatever the  `BulkOperationNode` type is).

Reply:
It's a GraphQL API and it's versioned, so we only get back the fields we ask for.

Reply:
Oy, I forgot we were dealing with GraphQL, I'll make the changes and push it.